src/platform/unix/mod.rs

```diff
@@ -471,8 +471,18 @@ impl OsIpcReceiverSet {
     pub fn select(&mut self) -> Result<Vec<OsIpcSelectionResult>,UnixError> {
         let mut selection_results = Vec::new();
         match self.poll.poll(&mut self.events, None) {
-            Ok(sz) if sz > 0 => {},
-            _ => { return Err(UnixError::last()); }
+            Ok(sz) => {
+                if sz < 1 {
+                    return self.select();
+                }
+            },
+            Err(ref e) => {
+                if e.kind() == ErrorKind::Interrupted {
+                    return self.select();
```

Comment:
Sneaky ;-)

I think I'd actually be more comfortable with an explicit loop -- not sure how others feel about this...